### PR TITLE
UW-255: Changes needed for SRW fill_jinja_template replacement.

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
**Description**

While integrating uwtools into SRW, a few changes were needed. 
- The arguments when generating the templates should point to default values, so I removed those.
- The name of the called script is now included in the logged information since it's helpful in the output in the log for SRW.

In addition, running the tests on Hera highlighted that we don't yet have jsonschema installed/supported there, so we should skip that test when the package isn't available.

**Type of change**

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

I ran the linter and pytests manually on hera.

**Checklist**

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation.  I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Review by WeirAE — COMMENTED** [review]
Tests clear, good change, just checking one thing - is the log entry in SRW not capturing the script name because of our log name change or is the SRW script not capturing the child call? We may want to change other logging to ensure the entries are still showing the correct script

File: src/uwtools/j2template.py
Comment (venitahagerty) [inline src/uwtools/j2template.py:1]: I'm unclear what changes happen when these two options (trim_blocks and lstrip_blocks) are not set to true, and if that would change anything in our testing.

**Comment by christinaholtNOAA** [comment]
@WeirAE SRW doesn't really have consistent logging throughout. There is a logger at the top level of the generate workflow script, but it isn't named or passed to any of the calls to other functions. There are no classes currently, so we're in a bit of a gray area. 

In this way, we can have the output that is sent to either a basic bash log, or a python log show up with some additional information. For example, this function is called separate three times for three separate config files from the bash run script for the forecast model.

File: src/uwtools/j2template.py
Comment (WeirAE) [reply src/uwtools/j2template.py:1]: trim_blocks removes the newline after a new block and lstrip_blocks strips leading spaces - I tested these a little and it didn't seem to affect the results and the tests seem to pass at least here...perhaps we should add another test to ensure this is being handled as expected?

Comment (christinaholtNOAA) [reply src/uwtools/j2template.py:1]: Removing these lines allows us to keep the same output structure as the template file we provide as input.

We likely want to open this method up enough that the user could potentially pass any additional native flags they'd like, but for now, I think going with the defaults is likely the short-term path forward.

**Review by WeirAE — APPROVED** [review]
Should be fine for now, we can revisit taking over logging for other apps later. LGTM

**Review by venitahagerty — APPROVED** [review]
LGTM

**Comment by christinaholtNOAA** [comment]
@venitahagerty Did I address your concerns? Could I do more?

**Comment by venitahagerty** [comment]
I'm good

**Pull request merged**